src/Language/PureScript/Parser/Types.hs

```diff
@@ -70,7 +70,7 @@ parseTypeAtom = indented *> P.choice
 
 parseConstrainedType :: TokenParser Type
 parseConstrainedType = do
-  constraints <- P.try (return <$> parseConstraint) <|> parens (commaSep1 parseConstraint)
+  constraints <- return <$> parseConstraint
```

Comment:
I guess we should change `ConstrainedType` to only include a single `Constraint` while we're at it. This should not block anything in this PR though.